### PR TITLE
Clean up the Block APIs

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Make `Block` a placeholder class as the result for `Operator`

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java [hunk 1]
```diff
@@ -29,4 +41,41 @@ public interface BlockDocIdSet {
...
+  /**
+   * Returns the number of entries (SV value contains one entry, MV value contains multiple entries) scanned in the
+   * filtering phase. This method should be called after the filtering is done.
+   */
+  long getNumEntriesScannedInFilter();
+
+  /**
+   * For scan-based FilterBlockDocIdSet, pre-scans the documents and returns a non-scan-based FilterBlockDocIdSet.
+   */
+  default BlockDocIdSet toNonScanDocIdSet() {
```
Comment (walterddr) [inline pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java:54]: `toPreScannedDocIdSet`

File: pinot-core/src/main/java/org/apache/pinot/core/common/Block.java [hunk 2]
```diff
@@ -19,41 +19,7 @@
 package org.apache.pinot.core.common;
 
 /**
- *
- * A block represents a set of rows.A segment will contain one or more blocks
- * Currently, it assumes only one column per block. We might change this in
- * future
+ * A {@code Block} represents the data block returned by the {@link Operator}. Each operator can return multiple blocks.
```
Comment (walterddr) [inline pinot-core/src/main/java/org/apache/pinot/core/common/Block.java:22]:
```suggestion
 * A {@code Block} represents the data block returned by the {@link Operator}. Each operator can return multiple blocks.
 * A {@code Block} can contain one or more columns of data depending on the operator implementation.
```

File: pinot-core/src/main/java/org/apache/pinot/core/operator/valsets/TransformBlockValSet.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.core.operator.docvalsets;
+package org.apache.pinot.core.operator.valsets;
```
Comment (walterddr) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/valsets/TransformBlockValSet.java:19]: what's the motivation of this change, seems to me docitset and docvalueset gives me clearer understanding

**Review by walterddr — APPROVED** [review]
lgtm. minor comments on naming and javadoc

File: pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/RowBlock.java [hunk 4]
```diff
@@ -16,18 +16,29 @@
...
+
+import java.util.List;
+import org.apache.pinot.common.utils.DataSchema;
+
 
 /**
- *
- * Not used for now, since most of the predicates make use of inverted index.
- * Revisit when we push predicate evaluation up into the operator
+ * The {@code RowBlock} contains a block of rows.
```
Comment (walterddr) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/RowBlock.java:26]:
```suggestion
 * The {@code RowBlock} contains a block of rows and a {@link DataSchema}.
 * each row contains a collection of entries which conformed with the DataSchema.
```

File: pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java:54]: I kind of prefer the current name (carried from the existing code) since we have scan-based and non-scan-based iterator, and we want to convert the scan-based one to non-scan-based

File: pinot-core/src/main/java/org/apache/pinot/core/common/Block.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/common/Block.java:22]: It can contain document ids, column values (seems always from multiple columns) or result rows. Updated the javadoc

File: pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/RowBlock.java (hunk 4 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/RowBlock.java:26]: Good suggestion. I've removed this class for now to make this PR pure clean up. Will re-introduce it in a separate PR
